qutip/core/qobj.py

```diff
@@ -1744,6 +1752,7 @@ def iscptp(self):
     def isherm(self):
         if self._isherm is not None:
             return self._isherm
+        self.tidyup()
```

Comment:
I feel `isherm` should not call `.tidyup()` -- I would be very surprised to discover that a method called `isX` changed the underlying data of a `Qobj`.

Reply:
`_isherm_csr_full` does not work without it, but it's a bug, not a choice for faster computation. It should totally be fixed there.